### PR TITLE
Render form dynamically according to form fields from kratos

### LINKED ISSUE #13: Error messages are not shown during registration
When creating an account and using an invalid password like `123456`, the UI will not show any error message and instead just show the vanilla registration screen. The data and error message however are available!

This should be fixed and the error message should be shown. This should work for both login and registration.

### PR DESCRIPTION
**Pull request by zepatrik**
closes #13

### CODE REVIEW

**Comment by zepatrik** [comment]
Screenshot from 2020-01-21 12-08-24

we need some better design for this

File: src/translations.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import {ExtendedFormField} from "types";
+
+const translations = {
+  password: {
+    title: 'Password',
+    order: 2
```
Comment (aeneasr) [inline src/translations.ts]: Let's call this `position`

File: src/translations.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    order: 0
+  }
+}
+
+type Translations = typeof translations
+
+const getTitle = (key: string, fallbackValue: string): string =>
+  key in translations ? translations[(key as keyof Translations)].title : fallbackValue
+
+const getOrder = (field: ExtendedFormField) =>
```
Comment (aeneasr) [inline src/translations.ts]: I would suggest `getPosition`

File: views/registration.hbs [hunk 3]
```diff
@@ -3,24 +3,13 @@
         <h1>{{projectName}}</h1>
         <h2>Sign up</h2>
 
-        <div class="kratos-errors">
-            {{#each errors}}
-                <p>{{ message }}</p>
-            {{/each}}
-        </div>
+        {{> kratosErrors}}
```
Comment (aeneasr) [inline views/registration.hbs]: I would rename this to `formErrors`

File: src/auth.ts [hunk 4]
```diff
@@ -40,23 +41,27 @@ export const authHandler = (type: 'login' | 'registration') => (
...
             },
           },
         },
       } = request
 
+      // inject handlebars helper keys
+      const formFields = Object.values(fields).map<ExtendedFormField>(field => ({
+        ...field,
+        ...{
+          isHidden: field.type === 'hidden',
```
Comment (aeneasr) [inline src/auth.ts]: As discussed, I think it would be better to move this to a partial with fallback

File: src/auth.ts [hunk 5]
```diff
@@ -40,23 +41,27 @@ export const authHandler = (type: 'login' | 'registration') => (
...
+          isPassword: field.type === 'password',
+          title: getTitle(field.name, field.name)
```
Comment (aeneasr) [inline src/auth.ts]: This should probably be a view helper instead

**Comment by zepatrik** [comment]
done @aeneasr

File: src/index.ts [hunk 6]
```diff
@@ -7,75 +7,83 @@ import debug from './debug'
...
+  handlebars({
+    extname: 'hbs',
+    layoutsDir: `${__dirname}/../views/layouts/`,
+    partialsDir: `${__dirname}/../views/partials/`,
+    defaultLayout: 'main',
+    helpers: {
+      json: (context: any) => JSON.stringify(context),
+      jsonPretty: (context: any) => JSON.stringify(context, null, 2),
+      getTitle,
+      whichFormInput: (type: string) =>
```
Comment (aeneasr) [inline src/index.ts]: Add a small comment to explain why we need this. I also think that a switch statement would be more readable. Remember that we're writing this as a reference for other developers - the more readable it is, the better!

Comment (aeneasr) [reply src/index.ts]: I'd also suggest renaming this to `toFormFieldTemplateName` or something along those lines

**Comment by aeneasr** [comment]
> Add a small comment to explain why we need this. I also think that a switch statement would be more readable. Remember that we're writing this as a reference for other developers - the more readable it is, the better!

Please address this comment

**Comment by aeneasr** [comment]
Awesome!

**Comment by aeneasr** [comment]
Awesome!

**Pull request merged**